protocol/requests/restore_account.go

```diff
@@ -0,0 +1,20 @@
+package requests
+
+import (
+	"errors"
+)
+
+var ErrRestoreAccountInvalidMnemonic = errors.New("restore-account: invalid mnemonic")
+
+type RestoreAccount struct {
+	Mnemonic string `json:"mnemonic"`
+	CreateAccount
+}
+
+func (c *RestoreAccount) Validate() error {
+	if len(c.Mnemonic) == 0 {
```

Comment:
It'd be good if we could have some tests for `restore_account#Validate` and `create_account#Validate`, maybe not in this PR if it's more urgent to merge the fix.